lib/rules/no-identical-ternary-expressions.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview Rule to flag ternary operators that have identical left and right expressions
+ * @author Che Fisher
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "disallow ternary operators with identical left and right hand expressions",
+            category: "Possible Errors",
+            recommended: true,
+            url: "https://eslint.org/docs/rules/no-identical-ternary-expressions"
+        },
+
+        schema: [],
+
+        messages: {
+            expressions: "Identical left and right hand expressions '{{expression}}'.",
+            conditions: "Identical test conditions encountered '{{condition}}'."
+        }
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+        const conditions = [];
+
+        /**
+         * Checks if the consequent and alternate expressions are identical, stripping out all whitespace first for non literals
+         *
+         * @param  {string} first a string representation of an expression
+         * @param  {string} second a string representation of an expression
+         * @param  {boolean} literal true if node type is a string literal
+         * @returns {boolean} True if the expressions are logically identical
+         * @private
+         */
+        function identicalExpressions(first, second, literal) {
```

Comment:
hmm... should this perhaps be added to the sourceCode helpers? I could imagine making it a little more general and have it instead compare two nodes, taking as parameters `identicalProperties(node1, node2, property)`